File: src/adapters/copilot-cli.ts
import type { HarnessAdapter, PluginMetadata } from "../domain/harness.js";
import {
  buildBaseManifest,
  buildPortableAgentArtifact,
  camelCaseHooks,
} from "./_shared.js";

const COPILOT_MANIFEST_KEYS = [
  "agents",
  "skills",
  "commands",
  "hooks",
  "mcpServers",
] as const;

function buildCopilotManifest(
  metadata: PluginMetadata,
  componentPaths: Parameters<typeof buildBaseManifest>[1],
): Record<string, unknown> {
  return {
    ...buildBaseManifest(metadata, componentPaths, COPILOT_MANIFEST_KEYS),
    category: "development",
    strict: true,
  };
}

export const copilotCliAdapter: HarnessAdapter = {
  name: "copilot-cli",
  displayName: "GitHub Copilot CLI",
  outputRoot: ".copilot",
  agentDirectory: "agents",
  skillDirectory: "skills",
  defaultModel: "gpt-5",
  notes: [
    "Copilot CLI resolves plugin manifests from .claude-plugin/plugin.json as its fourth search path, so the same manifest shape serves both Claude Code and Copilot CLI installations.",
  ],
  manifestDir: ".claude-plugin",
  buildManifest: (metadata, componentPaths) =>
    buildCopilotManifest(metadata, componentPaths),
  mcpFileName: ".mcp.json",
  buildHookConfig: (portable) => ({
    version: 1,
    hooks: camelCaseHooks(portable),
  }),
  buildAgentArtifact: buildPortableAgentArtifact,
  capabilities: {
    skillFrontmatterKeys: new Set<string>(),
    agentFrontmatterKeys: new Set<string>(),
    hookEvents: new Set(["sessionStart", "preToolUse", "postToolUse"]),
    toolNames: new Set<string>(),
    bootstrapHook: true,
  },
};

File: python/cheese_flow/adapters/copilot_cli.py
"""Port of `src/adapters/copilot-cli.ts`."""

from __future__ import annotations

from typing import Any

from cheese_flow.adapters._shared import (
    build_base_manifest,
    build_portable_agent_artifact,
    camel_case_hooks,
)
from cheese_flow.lib.harness import (
    HarnessAdapter,
    HarnessCapabilities,
    ManifestComponentPaths,
    PluginMetadata,
    PortableHooks,
)

COPILOT_MANIFEST_KEYS: tuple[str, ...] = (
    "agents",
    "skills",
    "commands",
    "hooks",
    "mcpServers",
)


def _build_manifest(
    metadata: PluginMetadata,
    component_paths: ManifestComponentPaths,
) -> dict[str, Any]:
    base = build_base_manifest(metadata, component_paths, COPILOT_MANIFEST_KEYS)
    return {**base, "category": "development", "strict": True}


def _build_hook_config(portable: PortableHooks) -> dict[str, Any]:
    return {"version": 1, "hooks": camel_case_hooks(portable)}


copilot_cli_adapter = HarnessAdapter(
    name="copilot-cli",
    displayName="GitHub Copilot CLI",
    outputRoot=".copilot",
    agentDirectory="agents",
    skillDirectory="skills",
    defaultModel="gpt-5",
    notes=(
        "Copilot CLI resolves plugin manifests from .claude-plugin/plugin.json as its fourth search path, so the same manifest shape serves both Claude Code and Copilot CLI installations.",  # noqa: E501
    ),
    manifestDir=".claude-plugin",
    buildManifest=_build_manifest,
    mcpFileName=".mcp.json",
    buildHookConfig=_build_hook_config,
    buildAgentArtifact=build_portable_agent_artifact,
    capabilities=HarnessCapabilities(
        skillFrontmatterKeys=frozenset(),
        agentFrontmatterKeys=frozenset(),
        hookEvents=frozenset({"sessionStart", "preToolUse", "postToolUse"}),
        toolNames=frozenset(),
        bootstrapHook=True,
    ),
)
